src/components/chat/chatInput.tsx

```diff
@@ -0,0 +1,55 @@
+'use client';
+import React from 'react';
+
+import { useUIState, useActions, useAIState } from 'ai/rsc';
+import { nanoid } from 'nanoid';
+
+import { AI, Actions } from '@/lib/actions';
+import { focuses } from '@/models/focus';
+
+import ChatBoxForm from './chatBoxForm';
+import DefaultQuestions from './defaultQuestions';
+import FocusBar from './focusBar';
+import { UserMessage } from './userMessage';
+
+// Container for all of components that can be used to send messages to the chat
+// Will send the actual message to the chatAI system
+const ChatInput = () => {
+  const [focus, setFocus] = React.useState(focuses[0]);
+
+  const [_, setMessagesUI] = useUIState<typeof AI>();
+  const [__, setAIState] = useAIState<typeof AI>();
```

Comment:
@srkirkland instead of setting the AI state here, you should do it inside the server action and use `getMutableAIState()`. and then add focus to `aiState.update()`. i'm not 100% sure what `setAIState` does here, but i'd guess it's just changing the context state?